docs/en/integrations/data-ingestion/data-formats/csv-tsv.md

```diff
@@ -0,0 +1,377 @@
+
+
+## Exporting to CSV
+
+Any format in our previous examples can also be used to export data. To export data from a table (or a query) to a CSV format, we use the same `FORMAT` clause:
+
+
+```sql
+SELECT *
+FROM sometable
+LIMIT 5
+FORMAT CSV;
+
+"Akiba_Hebrew_Academy","2017-08-01",241
+"Aegithina_tiphia","2018-02-01",34
+"1971-72_Utah_Stars_season","2016-10-01",1
+"2015_UEFA_European_Under-21_Championship_qualification_Group_8","2015-12-01",73
+"2016_Greater_Western_Sydney_Giants_season","2017-05-01",86
+```
+
+To add a header to the CSV file, we use the [CSVWithNames](/docs/en/interfaces/formats.md/#csvwithnames) format:
+
+```sql
+SELECT *
+FROM sometable
+LIMIT 5
+FORMAT CSVWithNames;
+
+"path","month","hits"
+"Akiba_Hebrew_Academy","2017-08-01",241
+"Aegithina_tiphia","2018-02-01",34
+"1971-72_Utah_Stars_season","2016-10-01",1
+"2015_UEFA_European_Under-21_Championship_qualification_Group_8","2015-12-01",73
+"2016_Greater_Western_Sydney_Giants_season","2017-05-01",86
+```
+
+
+### Saving exported data to a CSV file
+
+To save exported data to a file, we can use the [INTO…OUTFILE](/docs/en/sql-reference/statements/select/into-outfile.md) clause:
+
+```sql
+SELECT *
+FROM sometable
+INTO OUTFILE 'out.csv'
+FORMAT CSVWithNames;
+
+36838935 rows in set. Elapsed: 1.304 sec. Processed 36.84 million rows, 1.42 GB (28.24 million rows/s., 1.09 GB/s.)
+```
+
+Note how it took ClickHouse **~1** second to save 36m rows to a CSV file.
+
+
+### Exporting CSV with custom delimiters
+
+If we want to have other than comma delimiters, we can use the [format_csv_delimiter](/docs/en/operations/settings/settings.md/#format_csv_delimiter) settings option for that:
+
+```sql
+SET format_csv_delimiter = '|'
+```
+
+Now ClickHouse will use `|` as a delimiter for CSV format:
+
+```sql
+SELECT *
+FROM sometable
+LIMIT 5
+FORMAT CSV;
+
+"Akiba_Hebrew_Academy"|"2017-08-01"|241
+"Aegithina_tiphia"|"2018-02-01"|34
+"1971-72_Utah_Stars_season"|"2016-10-01"|1
+"2015_UEFA_European_Under-21_Championship_qualification_Group_8"|"2015-12-01"|73
+"2016_Greater_Western_Sydney_Giants_season"|"2017-05-01"|86
+```
+
+
+### Exporting CSV for Windows
+
+If we want a CSV file to work fine in a Windows environment, we should consider enabling [output_format_csv_crlf_end_of_line](/docs/en/operations/settings/settings.md/#output_format_csv_crlf_end_of_line) option. This will use `\r\n` as a line breaks instead of `\n`:
+
+```sql
+SET output_format_csv_crlf_end_of_line = 1;
+```
+
+## Detecting data types in a CSV file
+
+We might work with unknown CSV files in many cases, so we have to explore which types to use for columns. Clickhouse, by default, will try to guess data formats based on its analysis of a given CSV file. Detected data types can be explored using the `DESCRIBE` statement in pair with the [file()](/docs/en/sql-reference/table-functions/file.md) function:
```

Comment:
mention this is known as "Schema Inference"

Reply:
added mention